lib/default.nix

```diff
@@ -393,4 +393,96 @@ in {
   # one in the list will be used.
   combineFlakes = sep: prefixAndFlakes: builtins.foldl' addFlakes {}
     (lib.mapAttrsToList (prefix: flake: prefixFlake prefix sep flake) prefixAndFlakes);
+
+  mkFlake = project: packages: coverageProjectModule:
+    let
+      packageNames = project: builtins.attrNames (project.args.flake.packages project.hsPkgs);
+      checkedProject = project.appendModule { checkMaterialization = true; };
+      coverageProject = project.appendModule [
+        coverageProjectModule
+        {
+          modules = [{
+            packages = lib.genAttrs (packageNames project)
+              (_: { doCoverage = lib.mkDefault true; });
+          }];
+        }
+      ];
+      checks = builtins.listToAttrs (
+        lib.concatMap (packageName:
+          let package = project.hsPkgs.${packageName};
+          in lib.mapAttrsToList (n: v:
+              { name = "${packageName}:test:${n}"; value = v; })
+            (lib.filterAttrs (_: v: lib.isDerivation v) (package.checks))
+        ) (packageNames project));
+      ciJobs = {
```

Comment:
Why not go further: let's make `ciJobs :: Project -> ...` a top-level function too, so you can build a CI jobset from your project and put it wherever you like!

Reply:
Added `mkFlakeCiJobs`